be/src/env/env_hdfs.h

```diff
@@ -2,78 +2,95 @@
 
 #pragma once
 
-#include <hdfs/hdfs.h>
-
 #include "env/env.h"
-#include "object_store/s3_client.h"
 
 namespace starrocks {
 
-struct HdfsFsHandle {
-    enum class Type { LOCAL, HDFS, S3 };
-    Type type;
-    std::string namenode;
-    hdfsFS hdfs_fs;
-    S3Client* s3_client;
-};
-
-// class for remote read hdfs file
-// Now this is not thread-safe.
-class HdfsRandomAccessFile : public RandomAccessFile {
+// NOTE: Many methods in this class are unimplemented
+// TODO: Do NOT export the class definition, i.e, move the class
+//// definition to cpp file.
+class EnvHdfs : public Env {
 public:
-    HdfsRandomAccessFile(hdfsFS fs, const std::string& file_name, size_t file_size, bool usePread);
-    virtual ~HdfsRandomAccessFile() noexcept;
-    void close();
-    Status open();
-    Status read(uint64_t offset, Slice* res) const override;
-    Status read_at(uint64_t offset, const Slice& res) const override;
-    Status readv_at(uint64_t offset, const Slice* res, size_t res_cnt) const override;
-
-    Status size(uint64_t* size) const override {
-        *size = _file_size;
-        return Status::OK();
+    EnvHdfs() {}
+    ~EnvHdfs() override = default;
+
+    EnvHdfs(const EnvHdfs&) = delete;
+    void operator=(const EnvHdfs&) = delete;
+    EnvHdfs(EnvHdfs&&) = delete;
+    void operator=(EnvHdfs&&) = delete;
+
+    StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const std::string& path) override;
+
+    StatusOr<std::unique_ptr<RandomAccessFile>> new_random_access_file(const RandomAccessFileOptions& opts,
+                                                                       const std::string& path) override;
+
+    StatusOr<std::unique_ptr<SequentialFile>> new_sequential_file(const std::string& path) override {
```

Comment:
Should you put this code into default implementation? It's quite laboring to write such biolerplate code.

Reply:
> Should you put this code into default implementation? It's quite laboring to write such biolerplate code.

Yes, you are right. Maybe I can submit another PR to do this?

Reply:
Yes, please fix in following PR ASAP. I can not stand with long biolerplate code.